src/Workspaces/Core/Portable/Remote/RemoteArguments.cs

```diff
@@ -132,46 +126,35 @@ public static SerializableSymbolAndProjectId Create(ISymbol symbol, Project proj
     }
 
     [DataContract]
-    internal readonly struct SerializableReferenceLocation
+    internal readonly struct SerializableReferenceLocation(
+        DocumentId document,
+        SerializableSymbolAndProjectId? alias,
+        TextSpan location,
+        bool isImplicit,
+        SymbolUsageInfo symbolUsageInfo,
+        ImmutableDictionary<string, string> additionalProperties,
+        CandidateReason candidateReason)
     {
         [DataMember(Order = 0)]
-        public readonly DocumentId Document;
+        public readonly DocumentId Document = document;
 
         [DataMember(Order = 1)]
-        public readonly SerializableSymbolAndProjectId? Alias;
+        public readonly SerializableSymbolAndProjectId? Alias = alias;
 
         [DataMember(Order = 2)]
-        public readonly TextSpan Location;
+        public readonly TextSpan Location = location;
 
         [DataMember(Order = 3)]
-        public readonly bool IsImplicit;
+        public readonly bool IsImplicit = isImplicit;
 
         [DataMember(Order = 4)]
-        public readonly SymbolUsageInfo SymbolUsageInfo;
+        public readonly SymbolUsageInfo SymbolUsageInfo = symbolUsageInfo;
 
         [DataMember(Order = 5)]
-        public readonly ImmutableDictionary<string, string> AdditionalProperties;
+        public readonly ImmutableDictionary<string, string> AdditionalProperties = additionalProperties;
 
         [DataMember(Order = 6)]
-        public readonly CandidateReason CandidateReason;
-
-        public SerializableReferenceLocation(
-            DocumentId document,
-            SerializableSymbolAndProjectId? alias,
-            TextSpan location,
-            bool isImplicit,
-            SymbolUsageInfo symbolUsageInfo,
-            ImmutableDictionary<string, string> additionalProperties,
-            CandidateReason candidateReason)
-        {
-            Document = document;
-            Alias = alias;
-            Location = location;
-            IsImplicit = isImplicit;
-            SymbolUsageInfo = symbolUsageInfo;
-            AdditionalProperties = additionalProperties;
-            CandidateReason = candidateReason;
-        }
+        public readonly CandidateReason CandidateReason = candidateReason;
```

Comment:
Can we remove public field declarations and instead apply DataMember attributes to the generated fields?

Reply:
No.  There is currently no way to get to the "Generated Fields" for primary constructors currently (we don't even guarantee they are fields on the type).  There are future lang changes about being able to have a primary constructor, but be explicit if you want fields/props/etc.  but they are not in 12.

Reply:
You can't say `[field: DataMember(...)]` on the parameter?